dom.bs

```diff
@@ -1192,8 +1192,32 @@ for discussion).
    <li><p>If <var>event</var>'s <a>stop immediate propagation flag</a> is set, terminate the
    <a>invoke</a> algorithm.
 
-   <li><p>If <var>event</var>'s {{Event/type}} attribute value is not <var>listener</var>'s
-   <b>type</b>, terminate these substeps (and run them for the next <a>event listener</a>).
+   <li><p>Let <var>original event type</var> be <var>event</var>'s {{Event/type}} attribute value.
+
+   <li>
+    <p>If <var>original event type</var> is not <var>listener</var>'s <b>type</b>, run these
+    subsubsteps:
+
+    <ol>
+     <li>
+      <p>If <var>original event type</var> is an <a>ASCII case-insensitive</a> match for any of the
+      strings in the first column in the following table, set <var>event</var>'s {{Event/type}}
+      attribute value to the string in the second column on the same row as the matching string:
+
+      <table>
+       <thead>
+        <tr><th>Event Type<th>Legacy Event Type
+       <tbody>
+        <tr><td>"<code>animationend</code>"<td>"<code>webkitAnimationEnd</code>"
+        <tr><td>"<code>animationiteration</code>"<td>"<code>webkitAnimationIteration</code>"
+        <tr><td>"<code>animationstart</code>"<td>"<code>webkitAnimationStart</code>"
+        <tr><td>"<code>transitionend</code>"<td>"<code>webkitTransitionEnd</code>"
+      </table>
+
+     <li><p>If <var>event</var>'s {{Event/type}} attribute value is <var>listener</var>'s
+     <b>type</b>, continue, else terminate these substeps (and run them for the next
+     <a>event listener</a>).
```

Comment:
This logic seems wrong. There's a minor problem with "continue" not being a thing, but you could fix that by inverting the if logic. The bigger problem is that if you terminate the substeps, event's type will still be set to the new value, which might be incorrect for the next event listener.

It seems that whenever you terminate these substeps (and this happens not just here, but also later), event's type needs to be set back to the original event type.

Am I missing something?